x-pack/plugin/kql/build.gradle

```diff
@@ -17,7 +17,7 @@ base {
 
 dependencies {
   compileOnly project(path: xpackModule('core'))
-  compileOnly "org.antlr:antlr4-runtime:${versions.antlr4}"
+  api "org.antlr:antlr4-runtime:${versions.antlr4}"
```

Comment:
⚠️ Using `api` instead of `compileOnly`

Reply:
Using api will expose this dependency to other modules. Is that needed? 🤔

Reply:
Got the same question

Reply:
Honestly, this is the same way other plugins using ANTLR runtime (`lang-painless`, `ql,` and `esql-core`) include it, so I didn’t think too much about it.

Using api probably makes sense for `ql` and `esql-core` since they’re extended, while implementation should work fine for `kql` and `lang-painless`

Sorry, I saw your comment only after I merged the PR but I can change `api` to `implementation` in a follow-up if you want.

Reply:
An even better patch would be to add the ANTLR runtime to libs. Maybe something for later.